### DISCUSSION TITLE
Is there a 'redaction' functionality in Sulu?

### DISCUSSION THREAD

**Discussion by kwinne** [sulu/sulu#5819] [Q&A]
Hello,

I'm currently looking into Sulu for my next project(s) and I really like it, great job!
I was wondering if there is currently a redaction functionality in Sulu or if it is on the roadmap?

**To clarify what I mean, here is the use case:**
Users can create/modify pages / articles but can only save them as draft. Other users with extended roles can see if there are new drafts and can publish them when everything is okay.

Thanks in advance

**Comment by alexander-schranz** [comment]
Welcome @kwinne,

Good to hear you like sulu 🙂 

## The Current State

Currently if you have no permissions for `Live` you are not able to publish anything. So you can create 2 roles one which can publish things and one which can just create content.

Role without Publishing Permissions could look like this:

![Bildschirmfoto 2021-02-11 um 19 47 56](https://user-images.githubusercontent.com/1698337/107683615-0fc4f580-6ca2-11eb-9d92-6a5790134476.png)

There is currently no overview of only unpublished or draft pages so it would be needed that the content manager send a link manually to another user.

## Planned Feature

There is currently a whole publishing process planned to be implemented in sulu the workflow will look like the following:

```
        //                                                           unpublish
        //                   +--------------------------------------------------------------------------------------------+
        //                   |                                                                                            |
        //                   |                              publish                                                       |
        //                   |     +--------------------------------------------------------+                             |
        //                   |     |                                                        |                             |
        //                   |     |                       unpublish                        |           publish           |
        //                   |     |     +----------------------------------------------+   |   +---------------------+   |
        //                   V     |     V                                              |   V   V                     |   |
        // +-----+          +-------------+  request for review  +--------+           +------------+  remove draft  +-------+  request draft for review   +---------------+
        // |     |  create  |             |--------------------->|        |  publish  |            |<---------------|       |---------------------------->|               |
        // | New |--------->| Unpublished |                      | Review |---------->| Published  |                | draft |                             | Review draft  |
        // |     |          |             |<---------------------|        |           |            |--------------->|       |<----------------------------|               |
        // +-----+          +-------------+       reject         +--------+           +------------+  create draft  +-------+        reject draft         +---------------+
        //                                                                              A   |    A                                                                |
        //                                                                              +---+    |                          publish                               |
        //                                                                             publish   +----------------------------------------------------------------+
```

There should be then also an additional thread based discussion available for this "Request for Publishing" Workflow (little bit similar how Pull Requests in Github works). This is one of the feature we are currently working on. Please feel free to add your thoughts to the issue or give some details how you think the workflow should work for you.

[Request for Publishing Issue](https://github.com/sulu/sulu/issues/5233)

**Reply by kwinne** [reply]
Hello,

Thank you for this detailed answer, the flow you describe looks great and is exactly what I'm looking for. 
Just to be curious, is there a rough release date known for this functionality?

**Reply by niklasnatter** [reply]
We plan to ship a `request for publication` functionality with Sulu 2.4. There is no exact release date, but we intend to release a new minor version every 6 months. That means that the rough ETA for the feature is about one year from now 🙂

**Reply by alexander-schranz** [reply]
closing this discussion as there exists a issue: https://github.com/sulu/sulu/issues/5233 there was some changes about the roadmap as other features where more requested by our partners.
